firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/LiveSession.kt

```diff
@@ -16,359 +16,474 @@
-      @Serializable
-      internal data class ServerContent(@SerialName("modelTurn") val modelTurn: Content.Internal)
+  /**
+   * Receives responses from the model for both streaming and standard requests.
+   *
+   * Call [close] to stop receiving responses from the model.
+   *
+   * @return A [Flow] which will emit [LiveContentResponse] from the model.
+   *
+   * @throws [SessionAlreadyReceivingException] when the session is already receiving.
+   * @see stopReceiving
+   */
+  public fun receive(): Flow<LiveContentResponse> {
+    if (startedReceiving.getAndSet(true)) {
+      throw SessionAlreadyReceivingException()
     }
 
-    fun toInternal() = Internal(Internal.ServerContent(modelTurn))
-  }
+    // TODO(b/410059569): Remove when fixed
+    return flow {
+        while (true) {
+          val response = session.incoming.tryReceive()
+          if (response.isClosed || !startedReceiving.get()) break
 
-  internal class MediaStreamingSetup(val mediaChunks: List<MediaData.Internal>) {
-    @Serializable
-    internal class Internal(val realtimeInput: MediaChunks) {
-      @Serializable internal data class MediaChunks(val mediaChunks: List<MediaData.Internal>)
-    }
-    fun toInternal() = Internal(Internal.MediaChunks(mediaChunks))
+          val frame = response.getOrNull()
+          frame?.let { frameToLiveContentResponse(it) }?.let { emit(it) }
+
+          yield()
+        }
+      }
+      .onCompletion { stopAudioConversation() }
+
+    // TODO(b/410059569): Add back when fixed
+    //    return session.incoming.receiveAsFlow().transform { frame ->
+    //      val response = frameToLiveContentResponse(frame)
+    //      response?.let { emit(it) }
+    //    }.onCompletion {
+    //      stopAudioConversation()
+    //    }
   }
 
-  internal data class ToolCallSetup(
-    val functionCalls: List<FunctionCallPart.Internal.FunctionCall>
-  ) {
+  /**
+   * Stops receiving from the model.
+   *
+   * If this function is called during an ongoing audio conversation, the model's response will not
+   * be received, and no audio will be played; the live session object will no longer receive data
+   * from the server.
+   *
+   * To resume receiving data, you must either handle it directly using [receive], or indirectly by
+   * using [startAudioConversation].
+   *
+   * @see close
+   */
+  // TODO(b/410059569): Remove when fixed
+  public fun stopReceiving() {
+    if (!startedReceiving.getAndSet(false)) return
 
-    @Serializable
-    internal class Internal(val toolCall: ToolCall) {
+    scope.cancel()
+    playBackQueue.clear()
 
-      @Serializable
-      internal data class ToolCall(val functionCalls: List<FunctionCallPart.Internal.FunctionCall>)
-    }
+    audioHelper?.release()
+    audioHelper = null
+  }
 
-    fun toInternal(): Internal {
-      return Internal(Internal.ToolCall(functionCalls))
-    }
+  /**
+   * Sends function calling responses to the model.
```

Comment:
Add a note so people using start/stop conversation don't try to use this and rely on the handler instead.

Reply:
Good point, done